### PR TITLE
Restructured Package

### PR DESCRIPTION
**Pull request by bbonf**
+ some performance improvements

@mhkuu please don't merge yet

### CODE REVIEW

File: perfectextractor/apps/extractor/base.py [hunk 1]
```diff
@@ -112,18 +143,8 @@ def collect_file_names(self, dir_name):
         return file_names
 
     def generate_results(self, dir_name):
-        results = []
-
-        for file_name in self.collect_file_names(dir_name):
-            result = self.process_file(file_name)
-
-            for extractor in self.other_extractors:
```
Comment (bbonf) [inline perfectextractor/apps/extractor/base.py:120]: @mhkuu I think this didn't work as expected previously. Is it an important feature?

Comment (mhkuu) [reply perfectextractor/apps/extractor/base.py:120]: No, this can go for the moment!

**Comment by bbonf** [comment]
I think this could be merged now

File: perfectextractor/apps/extractor/base.py [hunk 2]
```diff
@@ -68,22 +91,30 @@ def __init__(self, language_from, languages_to=None,
...
         # Other variables
         self.other_extractors = []
         self.alignment_xmls = dict()
 
-    def process_folder(self, dir_name):
+    def process_folder(self, dir_name, progress_cb=None, done_cb=None):
         """
         Creates a result file and processes each file in a folder.
         """
+        progress_total = len(self.collect_file_names(dir_name))
```
Comment (mhkuu) [inline perfectextractor/apps/extractor/base.py:104]: `collect_file_names` is called in `generate_results` as well, potentially save this to a variable?

File: perfectextractor/corpora/europarl/extractor.py [hunk 3]
```diff
@@ -114,21 +119,26 @@ def get_sentence(self, element):
...
-        if line:
-            return line[0]
-        else:
-            return None
+        return self._segment_by_id(tree, segment_number)
 
     def get_line(self, tree, segment_number):
-        line = tree.xpath('//s[@id="' + segment_number + '"]')
+        line = self._segment_by_id(tree, segment_number)
         if line:
```
Comment (mhkuu) [inline perfectextractor/corpora/europarl/extractor.py:142]: I get a warning here that `if line is not None` is preferred (like in the `get_line_and_pp` method below).

**Pull request merged**